File: frontend/src/services/noteService.js
import api from './api'
import toast from 'react-hot-toast'

export const noteService = {
  async createNote(noteData) {
    try {
      const response = await api.post('/notes', noteData)
      toast.success('Note created successfully!')
      return response.data
    } catch (error) {
      const message = error.response?.data?.detail || 'Failed to create note'
      toast.error(message)
      throw error
    }
  },

  async getNotes() {
    try {
      const response = await api.get('/notes')
      return response.data
    } catch (error) {
      const message = error.response?.data?.detail || 'Failed to fetch notes'
      toast.error(message)
      throw error
    }
  },

  async updateNote(noteId, noteData) {
    try {
      const response = await api.put(`/notes/${noteId}`, noteData)
      toast.success('Note updated successfully!')
      return response.data
    } catch (error) {
      const message = error.response?.data?.detail || 'Failed to update note'
      toast.error(message)
      throw error
    }
  },

  async deleteNote(noteId) {
    try {
      const response = await api.delete(`/notes/${noteId}`)
      toast.success('Note deleted successfully!')
      return response.data
    } catch (error) {
      const message = error.response?.data?.detail || 'Failed to delete note'
      toast.error(message)
      throw error
    }
  }
}

File: backend/app/services/note_service.py
# app/services/note_service.py
from app.db.database import note_collection
from fastapi import HTTPException
from bson import ObjectId

async def create_note(user_id: str, data):
    note = {
        "title": data.title,
        "content": data.content,
        "images": data.images,   # ✅ added
        "videos": data.videos,   # ✅ added
        "owner_id": user_id
    }

    result = await note_collection.insert_one(note)
    note["_id"] = str(result.inserted_id)

    return note

async def get_notes(user_id: str):
    notes = []
    cursor = note_collection.find({"owner_id": user_id})

    async for note in cursor:
        note["_id"] = str(note["_id"])
        notes.append(note)

    return notes


async def update_note(note_id: str, user_id: str, data):
    note = await note_collection.find_one({"_id": ObjectId(note_id)})

    if not note or note["owner_id"] != user_id:
        raise HTTPException(status_code=403, detail="Unauthorized")

    update_data = data.dict(exclude_unset=True)

    await note_collection.update_one(
        {"_id": ObjectId(note_id)},
        {"$set": update_data}
    )

    return {"message": "Updated"}

async def delete_note(note_id: str, user_id: str):
    note = await note_collection.find_one({"_id": ObjectId(note_id)})

    if not note or note["owner_id"] != user_id:
        raise HTTPException(status_code=403, detail="Unauthorized")

    await note_collection.delete_one({"_id": ObjectId(note_id)})

    return {"message": "Deleted"}